api/s2n.h

```diff
@@ -596,6 +596,210 @@ extern int s2n_cert_get_utf8_string_from_extension_data_length(const uint8_t *ex
 S2N_API 
 extern int s2n_cert_get_utf8_string_from_extension_data(const uint8_t *extension_data, uint32_t extension_len, uint8_t *out_data, uint32_t *out_len);
 
+/* PSK Hash Algorithm - RFC 8446 Section-2.2 */
+/* https://datatracker.ietf.org/doc/html/rfc8446#section-2.2*/
+typedef enum {
+    S2N_PSK_HMAC_SHA256,
+    S2N_PSK_HMAC_SHA384,
+} s2n_psk_hmac;
+
+struct s2n_psk;
+
+/**
+ * Creates a new s2n external PSK object with `S2N_PSK_HMAC_SHA256` as the default PSK hash algorithm.
```

Comment:
We might want to write out "pre-shared key" to be as clear as possible. We're very familiar with "PSK", but users might not be.